### ISSUE TITLE
Cannot import some pictures : error 22P02

### ISSUE THREAD

**Issue by fariouche** [LycheeOrg/Lychee#1234]
### Detailed description of the problem [REQUIRED]
Some pictures cannot be uploaded. Seems to be all pictures from my smartphone.
This happens in both command line (lychee:sync) and web interface.

I'm using postgresql 14 and I haven't setup imagick yet.
As far as I can tell, there are floats in the metadata and the sql request is chocking expecting an integer

I hope this can be easily solved, so that I can start using it. Keep the good work! :+1: 

### Steps to reproduce the issue
**Steps to reproduce the behavior:**
Just upload a picture but not sure this is related to this particular pictures taken by my smartphone.

### Output of the diagnostics  [REQUIRED]

```
    Diagnostics
    -------
    Warning: '/mnt/raid/lychee/dist/user.css' does not exist or has insufficient read/write privileges.
    Warning: Dropbox import not working. dropbox_key is empty.
    Warning: Pictures that are rotated lose their metadata! Please install Imagick to avoid that.

    System Information
    --------------
    Lychee Version (release):        4.4.0
    DB Version:                      4.4.0
    
    composer install:                --no-dev
    APP_ENV:                         production
    APP_DEBUG:                       true
    
    System:                          Linux
    PHP Version:                     8.1
    PHP User agent:                  Lychee/4 (https://lycheeorg.github.io/)
    Max uploaded file size:          16G
    Max post size:                   16G
    Max execution time:              3600
    PostgreSQL Version:              PostgreSQL 14.1 on x86_64-pc-linux-gnu, compiled by x86_64-pc-linux-gnu-gcc (Gentoo Hardened 11.2.1_p20220115 p4) 11.2.1 20220115, 64-bit
    
    Imagick:                         -
    Imagick Active:                  1
    Imagick Version:                 -
    GD Version:                      bundled (2.1.0 compatible)




    Config Information
    --------------
    version:                         040400
    check_for_updates:               0
    sorting_Photos_col:              taken_at
    sorting_Photos_order:            ASC
    sorting_Albums_col:              max_taken_at
    sorting_Albums_order:            ASC
    imagick:                         1
    skip_duplicates:                 0
    small_max_width:                 0
    small_max_height:                360
    medium_max_width:                1920
    medium_max_height:               1080
    lang:                            en
    layout:                          1
    image_overlay_type:              desc
    default_license:                 none
    compression_quality:             90
    full_photo:                      1
    delete_imported:                 0
    Mod_Frame:                       1
    Mod_Frame_refresh:               30
    thumb_2x:                        1
    small_2x:                        1
    medium_2x:                       1
    landing_page_enable:             0
    landing_owner:                   John Smith
    landing_title:                   John Smith
    landing_subtitle:                Cats, Dogs & Humans Photography
    landing_facebook:                https://www.facebook.com/JohnSmith
    landing_flickr:                  https://www.flickr.com/JohnSmith
    landing_twitter:                 https://www.twitter.com/JohnSmith
    landing_instagram:               https://instagram.com/JohnSmith
    landing_youtube:                 https://www.youtube.com/JohnSmith
    landing_background:              dist/cat.jpg
    site_title:                      Lychee v4
    site_copyright_enable:           1
    site_copyright_begin:            2019
    site_copyright_end:              2019
    additional_footer_text:          
    display_social_in_gallery:       0
    public_search:                   0
    SL_enable:                       0
    SL_for_admin:                    0
    public_recent:                   0
    recent_age:                      1
    public_starred:                  0
    downloadable:                    0
    photos_wraparound:               1
    map_display:                     0
    zip64:                           1
    map_display_public:              0
    map_provider:                    Wikimedia
    force_32bit_ids:                 0
    map_include_subalbums:           0
    update_check_every_days:         3
    has_exiftool:                    0
    share_button_visible:            0
    import_via_symlink:              0
    has_ffmpeg:                      0
    location_decoding:               0
    location_decoding_timeout:       30
    location_show:                   1
    location_show_public:            0
    rss_enable:                      0
    rss_recent_days:                 7
    rss_max_items:                   100
    prefer_available_xmp_metadata:   0
    editor_enabled:                  1
    lossless_optimization:           0
    swipe_tolerance_x:               150
    swipe_tolerance_y:               250
    local_takestamp_video_formats:   .avi|.mov
    log_max_num_line:                1000
    unlock_password_photos_with_url_param: 0
    nsfw_visible:                    1
    nsfw_blur:                       0
    nsfw_warning:                    0
    nsfw_warning_admin:              0
    map_display_direction:           1
    album_subtitle_type:             oldstyle
    upload_processing_limit:         4
    public_photos_hidden:            1
    new_photos_notification:         0
```
### Browser and system

This is the relevant log part:

```
2022-02-28 18:16:18 -- error   -- App\Actions\Photo\Extensions\Save::recover -- 59 -- Something went wrong, error 22P02, SQLSTATE[22P02]: Invalid text representation: 7 ERROR:  invalid input syntax for type integer: "607.5"
CONTEXT:  unnamed portal parameter $32 = '...' (SQL: insert into "photos" ("id", "checksum", "title", "url", "description", "tags", "width", "height", "type", "filesize", "iso", "aperture", "make", "model", "lens", "shutter", "focal", "taken_at", "taken_at_orig_tz", "latitude", "longitude", "altitude", "imgDirection", "location", "livePhotoContentID", "public", "star", "album_id", "owner_id", "thumb2x", "thumbUrl", "medium_width", "medium_height", "medium2x_width", "medium2x_height", "small_width", "small_height", "small2x_width", "small2x_height", "updated_at", "created_at") values (16460721747793, 76085e58611e0d9a321115b322d07f238d34b358, 20170414_223757(0), 76085e58611e0d9a321115b322d07f23.jpg, , , 2268, 4032, image/jpeg, 4092045, 200, f/1.7, samsung, SM-G935F, , 1/10 s, 4 mm, 2017-04-14 22:37:57, UTC, 36.718333333333, 10.367777777778, 37, ?, ?, ?, 0, 0, ?, 0, 1, 76085e58611e0d9a321115b322d07f23.jpeg, 607.5, 1080, 1215, 2160, 202.5, 360, 405, 720, 2022-02-28 18:16:18, 2022-02-28 18:16:18) returning "id")
    2022-02-28 18:16:15 -- notice  -- App\Actions\Photo\Extensions\ImageEditing::createThumb -- 165 -- Photo URL is 76085e58611e0d9a321115b322d07f23.jpg
    2022-02-28 18:16:14 -- notice  -- App\Models\Extensions\ConfigsHas::hasImagick -- 19 -- hasImagick : false
```

**Comment by d7415** [comment]
It looks like it's `medium_width`, which should definitely be an integer.

**Comment by fariouche** [comment]
What is medium_width? I don't see this field in exif format, and darktable and gimp do not show this information.

**Comment by d7415** [comment]
Sorry, that was more for the rest of the team. `medium_width` is generated by Lychee, and should be an integer.

**Comment by kamil4** [comment]
I'm seeing it on current `master` as well... Using ImageMagick seems to prevent it, as does using MySQL (which silently rounds the numbers to full integers).

**Comment by kamil4** [comment]
@fariouche I have a fix pending in #1235. It's against the current `master` but by the look of it it should apply cleanly to 4.4.0. You simply need to update three lines in the file `Lychee/app/Image/GdHandler.php` and that's it.

**Comment by fariouche** [comment]
it is working! thanks!
